school_center.py

```diff
@@ -178,6 +178,52 @@ def is_allocated(scode1: str, scode2: str) -> bool:
     return allocations.get(scode1, {}).get(scode2) is not None
 
 
+def allocate_to_centers(centers: List[Dict[str, str]], school: Dict[str, str], to_allot: int, remaining_cap: Dict[str, int], stretch: bool=False):
+    allocated_centers = {}
+    per_center = calc_per_center(to_allot)
+    
+    for center in centers:
+        school_code = school['scode']
+        center_code = center['cscode']
+        writer.writerow([
+            school_code, 
+            school['count'], 
+            school['name-address'], 
+            school['lat'], 
+            school['long'], 
+            center_code, 
+            center['name'], 
+            center['address'], 
+            center['capacity'], 
+            center['distance_km']
+        ])
+
+        if is_allocated(center_code, school_code):
+            continue
+
+        centers_cap_left = remaining_cap[center_code]
+
+        if stretch:
+            stretched_capacity = math.floor(int(center['capacity']) * STRETCH_CAPACITY_FACTOR + centers_cap_left)
+            next_allot = min(to_allot, max(stretched_capacity, MIN_STUDENT_IN_CENTER))
+            centers_cap_left = stretched_capacity
+        else:
+            next_allot = min(to_allot, per_center, max(centers_cap_left, MIN_STUDENT_IN_CENTER))
+        
+        if to_allot > 0 and next_allot > 0 and centers_cap_left >= next_allot:
+            if next_allot < MIN_STUDENT_IN_CENTER and len(allocated_centers) > 0:
+                # get the first center code
+                allocate_center_code = centers[0]['cscode']
```

Comment:
first center may not always be the best choice to allocate remaining students. 
- it may not  have required capacity as we have used center_cap_left for current_center. 
- in my opinion assigning them to last allotted center might be preferable for registration number distribution. eg: school with 408 students - Case 1: Center A:   registration no 1 - 200; Center B reg no 201 - 408. Case 2: Center A: registration no. 1 - 208; Center B reg no 209 - 408.  Case 1 would be preferred as starting registration number for all centers is a easy number

Reply:
@sapradhan Thank you for your feedback. I agree with your point that the distributed number might not be **cleaner**. 

I think for the given scenario 2:
- Center A: The registration number will be from 1 - 200
- Center B: The registration number will be from 201 - 400

The remaining students ( 8 ) and their registration number will be allocated to the Center A making them:
- CenterA: The registration number from 1 - 200 & 401 to 408.

So, my question will be should we make the allocation clear or the registration number distribution. If the latter is the expected result then I can make the changes. Let me know.

Reply:
@sapradhan Thank you for your feedback. I agree with your point that the distributed number might not be **cleaner**. 

I have one question, so when assigning the registration number, we will not care about the offset as below:
- Center A: reg number from 1 to 200
- Center B: reg number from 201 to 400
- Remaining reg number ( 401 to 408 )

Let me know and I will make changes to assign to the last center.

Reply:
Handling issue 1 should be good enough for now. 

I realize now that issue 2 may not be that important.
Registration numbers distribution are not currently handled by this script. I only brought up registration numbers looking at [notice published by NEB](https://www.neb.gov.np/uploads/photos/VWVFiwAWTb240422082939.pdf), so may be part of a future enhancement. Also centers can be ordered differently so as to make the registration no. distribution cleaner. eg - if we allot Center A - 208, Center B - 200, it does not mean we have to put first 208 students to Center A. We can distribute Center B 1 - 200, Center A 201 - 408. So I take back my previous opinion that assigning students to last allotted center is preferred.